examples/pipelines/starcoder/pipeline.py

```diff
@@ -46,7 +48,20 @@
 pipeline.add_op(load_from_hub_op)
 pipeline.add_op(filter_line_length_op, dependencies=load_from_hub_op)
 pipeline.add_op(filter_comments_op, dependencies=filter_line_length_op)
-pipeline.add_op(pii_redaction_op, dependencies=load_from_hub_op)
+pipeline.add_op(pii_redaction_op, dependencies=filter_comments_op)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--local", action="store_true")
+    if parser.parse_args().local:
+        compiler = DockerCompiler()
+        extra_volumes = [
+            "$HOME/.config/gcloud/application_default_credentials.json:/root/.config/gcloud/application_default_credentials.json:ro"
+        ]
+        compiler.compile(pipeline=pipeline, extra_volumes=extra_volumes)
+        logger.info("Run `docker-compose up` to run the pipeline.")
 
-# compile
-client.compile_and_run(pipeline=pipeline)
+    else:
+        client = Client(host=PipelineConfigs.HOST)
+        client.compile_and_run(pipeline=pipeline)
```

Comment:
I would prefer to include this into the `Pipeline` class or a separate command before updating the example. Ideally, we don't need any code changes in this file to make it work with a different runner.

Reply:
Agreed, maybe as a different PR as I don't want to bloat this one